### PR TITLE
[1LP][RFR] Implementing changes to improve test cases

### PR DESCRIPTION
**Pull request by anurag03**
Purpose or Intent
======================
Implementing the suggested changes in the test case to improve it.

{{pytest: -v -k 'test_infra_tile_page_per_item or test_infra_list_page_per_item or test_infra_grid_page_per_item or test_infra_report_page_per_item' }}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 0ba53ff8936f901691c10ecb6d4c7703122049f0:

`cfme/tests/configure/test_visual_infra.py`:
- :red_circle: Line 8:1: **F401** *'cfme.intelligence.reports.reports.CannedReportView' imported but unused*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: cfme/tests/configure/test_visual_cloud.py [hunk 1]
```diff
@@ -100,13 +100,13 @@ def test_cloud_grid_page_per_item(request, page, set_grid, appliance):
...
-    navigate_to(page, 'All')
-    tb.select('Grid View')
-    from cfme.web_ui import paginator
-    if paginator.rec_total() is not None:
-        if int(paginator.rec_total()) >= int(limit):
-            assert int(paginator.rec_end()) == int(limit), \
-                "Gridview Failed for page {}!".format(page)
+    view = navigate_to(page, 'All')
+    view.toolbar.view_selector.select('Grid View')
+    min_item, max_item, item_amt = view.paginator.paginator.page_info()
```
Comment (rlbabyuk) [inline cfme/tests/configure/test_visual_cloud.py]: **Optional:** max_item_amount property can be added here, the same as for view.paginator.items_amount to reduce usage of view.paginator.paginator

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

File: cfme/infrastructure/virtual_machines.py [hunk 2]
```diff
@@ -1150,10 +1150,10 @@ class VmAllWithTemplates(CFMENavigateStep):
     def step(self):
         self.prerequisite_view.navigation.select('Compute', 'Infrastructure', 'Virtual Machines')
         self.view.sidebar.vmstemplates.tree.click_path('All VMs & Templates')
-        if self.view.pagination.is_displayed:
```
[the remark this answers is not in the document]
Comment (anurag03) [reply cfme/infrastructure/virtual_machines.py]: I checked, I haven't got any failure in any test case related to this change @rlbabyuk, @psav  suggested me these changes.

File: cfme/tests/configure/test_visual_cloud.py [hunk 3]
```diff
@@ -90,13 +90,13 @@ def test_cloud_grid_page_per_item(request, page, set_grid, appliance):
...
+    if view.paginator.items_amount is not None:
+        if int(view.paginator.items_amount) >= int(limit):
+            assert int(max_item) == int(limit), "Gridview Failed for page {}!".format(page)
```
Comment (quarckster) [inline cfme/tests/configure/test_visual_cloud.py]: **REQUIRED**
Here and below, please don't repeat these nested conditions.
```python
if view.paginator.items_amount is not None and int(view.paginator.items_amount) >= int(limit):
    assert int(max_item) == int(limit), "Gridview Failed for page {}!".format(page)
```

**Comment by anurag03** [comment]
Changing the state from [1LP][WIPTEST] to [1LP][RFR] as the failure is due to no record and, hence there is no paginator thus the test case is failing.
filedump-exception_screenshot

**Pull request merged**